common/util.cpp

```diff
@@ -163,4 +163,43 @@ bool isPunct(int c) {
 	return ispunct((byte)c);
 }
 
+
+#pragma mark -
+
+
+Common::String getHumanReadableBytes(uint64 bytes, Common::String &unitsOut) {
+	Common::String result = Common::String::format("%lu", bytes);
```

Comment:
Using `%lu` here might be an issue since on some systems a long int is 32 bits.
That being said it seems to me that this line, as well as lines 174 to 178, could just be removed.

Reply:
1) Should I use `%llu` instead?
2) If we remove these lines (guess not completely, but leaving `if (bytes >= 1024) bytes /= 1024;`), the smallest units this function would return would be megabytes. It's a little bit inconsistent and probably not really convinient for files less than 1 MB.

Reply:
Right. Indeed those lines cannot be fully removed.
Line 171 can be removed though, as well as line 176 (the two lines on which you format the result with a `%lu`) since result is always set again on line 201 from the float value.

By the way, is there a reason why you don't set the float before the first `if` case (and divide it in the if case) to also have one decimal after the dot in the kb case in your string? Here it's a bit inconsistent with kb always having no decimal and Mb, Gb, and Tb having one.

Reply:
Hmm, I didn't mean to return floating point on bytes and KBs; but I wanted to not repeat the same line three times (in each of those `if`) and put it afterwards.

So, the idea was that bytes (0..1023) could be printed as is; KBs are small enough, so there is no need of decimal part, but after that (in MBs, GBs, TBs) decimal part gets important. May be it's better to use decimal for KBs for consistency though.

Reply:
You are likely to still want to not use decimal point in the bytes case though, so the question on `%lu` remains.

One possibility would indeed be to use `%llu` and cast the `bytes` value to an `unsigned long long int` explicitly to avoid a warning in case `uint64` is defined as an `unsigned long int`. However `%llu` was added in the C99 standard, and I am wondering if there might still be compilers out there that does not support it (I think we might still be using some pretty old compilers for some ports).

Here since you are only going to use it if `bytes` is less than 1024 (or 1024 * 1024 if you use it for Kb as well), you know it fits in a 32 bits unsigned integer. So you could also downcast to `unsigned long int` and keep using `%lu`).

Reply:
I've commited updated version, that does downcast to `unsigned long int`, so I guess we can resolve this one?

Reply:
Yes, I am marking this as resolved.